radixdlt/src/integration/java/com/radixdlt/integration/distributed/MockedLedgerModule.java

```diff
@@ -40,14 +47,20 @@ public void configure() {
 	Ledger syncedLedger() {
 		return new Ledger() {
 			@Override
-			public LedgerHeader prepare(VerifiedVertex vertex) {
-				return LedgerHeader.create(
-					vertex.getParentHeader().getLedgerHeader().getEpoch(),
-					vertex.getView(),
-					vertex.getParentHeader().getLedgerHeader().getAccumulatorState(),
-					0L,
-					false
-				);
+			public Optional<PreparedVertex> prepare(LinkedList<PreparedVertex> previous, VerifiedVertex vertex) {
+				final long timestamp = vertex.getQC().getTimestampedSignatures().weightedTimestamp();
+				final LedgerHeader ledgerHeader = vertex.getParentHeader().getLedgerHeader()
+					.updateViewAndTimestamp(vertex.getView(), timestamp);
+
+				return Optional.of(vertex.withHeader(ledgerHeader).andCommands(
+					vertex.getCommand() != null ? ImmutableSet.of(vertex.getCommand()) : ImmutableSet.of(),
```

Comment:
The pattern 'value != null ? ImmutableSet.of(value) : ImmutableSet.of()' (or it's inverted version with '==') appears several times in code. Perhaps it worth to create utility method for that.

Reply:
Going through this again I dont think this is a pattern we will want to support in the long term so would rather not create a utility for it. The pattern only exists as a symptom of deeper structural issues which should get resolved in the near-term (specifically, support for batched commands).